### DISCUSSION TITLE
Wie kann ich einen String von Umlauten und Sonderzeichen befreien?

### DISCUSSION THREAD

**Discussion by tyrant88** [FriendsOfREDAXO/.github#31] [Q&A] [answered]
Ich möchte aus einem Bezeichner alle Umlaute und Sonderzeichen entfernen, damit ich ihn als Dateinamen im Medienpool verwenden kann.
`$fileName = 'wichtige-Datön&Fäkten.pdf'`
 
Und außerdem möchte ich diesen String noch als Sprungmarke (Anker) verwenden können.
Also z.B. `<a id="$anchor">; wobei $anchor = 'Sprüng-Märke';`

**Comment by eaCe** [comment, marked as answer]
Um Sonderzeichen oder Umlaute zu entfernen kannst du `rex_string::normalize()` nutzen:

```php
$fileName = rex_string::normalize('wichtige-Datön&Fäkten');
```

$fileName entspricht dann z.B. `wichtige_datoen_faekten`.

In deinem Beispiel z.B.:

```php
<?php
$anchor = rex_string::normalize('Sprüng-Märke');
?>

<a href="#<?= $anchor ?>">Anker</a>
<div id="<?= $anchor ?>"></div>
```

**Comment by skerbis** [comment]
Auch hier zu finden: https://friendsofredaxo.github.io/tricks/snippets/umlaute_umschreiben
